DeepSpeech.py

```diff
@@ -548,22 +577,83 @@ def train():
             init_op = tfv1.variables_initializer(missing_variables)
             session.run(init_op)
             loaded = True
-
-        tfv1.get_default_graph().finalize()
-
+            
         if not loaded and FLAGS.load in ['auto', 'last']:
+            tf.initialize_all_variables().run()
             loaded = try_loading(session, checkpoint_saver, 'checkpoint', 'most recent')
         if not loaded and FLAGS.load in ['auto', 'best']:
-            loaded = try_loading(session, best_dev_saver, 'best_dev_checkpoint', 'best validation')
-        if not loaded:
-            if FLAGS.load in ['auto', 'init']:
-                log_info('Initializing variables...')
-                session.run(initializer)
-            else:
-                log_error('Unable to load %s model from specified checkpoint dir'
-                          ' - consider using load option "auto" or "init".' % FLAGS.load)
-                sys.exit(1)
-
+            tf.initialize_all_variables().run()
+            loaded = try_loading(session, best_dev_saver, 'best_dev_checkpoint', 'best validation') 
+        if not loaded and FLAGS.load == "transfer":
+            if FLAGS.source_model_checkpoint_dir:
```

Comment:
All of this branch is very close to the CUDNN loading code we already have, is there some way we could factor a little bit, to avoid duplication? Or is there some tricy part that makes it not doable.